SonarQube.Client/SonarQubeService.cs

```diff
@@ -138,33 +144,29 @@ public async Task ConnectAsync(ConnectionInformation connection, CancellationTok
 
             try
             {
-                // We have to set IsConnected to true here, otherwise the two calls to InvokeRequestAsync
-                // below will fail. However, at this point we don't really know if the server is running
-                // so the web calls could fail.
-                IsConnected = true;
                 var serverTypeDescription = connection.IsSonarCloud ? "SonarCloud" : "SonarQube";
 
                 logger.Debug($"Getting the version of {serverTypeDescription}...");
 
-                var versionResponse = await InvokeRequestAsync<IGetVersionRequest, string>(token);
-                ServerInfo = new ServerInfo(Version.Parse(versionResponse),
-                    connection.IsSonarCloud ? ServerType.SonarCloud : ServerType.SonarQube);
+                var versionResponse = await InvokeUnauthenticatedRequestAsync<IGetVersionRequest, string>(request => { }, token);
+                var serverInfo = new ServerInfo(Version.Parse(versionResponse), connection.IsSonarCloud ? ServerType.SonarCloud : ServerType.SonarQube);
 
-                logger.Info($"Connected to {serverTypeDescription} '{ServerInfo.Version}'.");
+                logger.Info($"Connected to {serverTypeDescription} '{serverInfo.Version}'.");
 
                 logger.Debug($"Validating the credentials...");
 
-                var credentialResponse = await InvokeRequestAsync<IValidateCredentialsRequest, bool>(token);
+                var credentialResponse = await InvokeUnauthenticatedRequestAsync<IValidateCredentialsRequest, bool>(request => { }, token);
```

Comment:
I feel that `InvokeUnauthenticatedRequestAsync` is not the best name for it.
It's not unauthenticated, it's more "InvokeWithoutWaitingForConnection".

Reply:
`InvokeUncheckedRequestAsync`? And maybe renaming the other method `InvokeCheckedRequestAsync`.

Or some variation thereof: `InvokeRequestWithCheckAsync`.

`Checked` still isn't great, but it's slightly better than `Unauthenticated`.